instance/models/mixins/openedx_monitoring.py

```diff
@@ -42,28 +42,34 @@ def enable_monitoring(self):
             self.logger.warning('Skipping monitoring setup, '
                                 'NEWRELIC_ADMIN_USER_API_KEY not set')
             return
-        self.logger.info('Setting up New Relic Synthetics monitors')
-
-        # Delete existing monitors if they don't monitor this instance's
-        # public urls
-        monitors = newrelic.get_synthetics_monitors()
-        already_enabled = [monitor for monitor in monitors
-                           if monitor['uri'] in self._urls_to_monitor]
-        already_enabled_ids = {enabled['id'] for enabled in already_enabled}
-        for monitor in self.new_relic_availability_monitors.exclude(
-                pk__in=already_enabled_ids):
-            monitor.delete()
+        self.logger.info('Checking New Relic Synthetics monitors')
+
+        already_monitored_urls = set()
+
+        for monitor in self.new_relic_availability_monitors.all():
+            url = newrelic.get_synthetics_monitor(monitor.pk)['uri']
+            if url in self._urls_to_monitor:
```

Comment:
The set `_urls_to_monitor` is a property that gets evaluated in every call (and all its entries also come from evaluating properties, which depend on further properties).  In this case it does not really matter, but in general I would prefer to store this to a local variabl once and reuse it.  Keeping a bit of hygiene in this regard helps avoiding lots of performance problems in the long run.